frontend/src/components/application/AppVerificationSetup.tsx

```diff
@@ -0,0 +1,89 @@
+import { Disclosure } from "@headlessui/react"
+import { useTranslation } from "next-i18next"
+import {
+  FunctionComponent,
+  ReactElement,
+  useCallback,
+  useEffect,
+  useState,
+} from "react"
+import { verifyApp } from "src/asyncs/app"
+import { fetchVerificationAvailableMethods } from "src/fetchers"
+import { Appstream } from "src/types/Appstream"
+import { VerificationAvailableMethods } from "src/types/VerificationAvailableMethods"
+import Button from "../Button"
+import Spinner from "../Spinner"
+
+interface Props {
+  app: Appstream
+}
+
+const AppVerificationSetup: FunctionComponent<Props> = ({ app }) => {
+  const { t } = useTranslation()
+
+  const [verificationMethods, setVerificationMethods] =
+    useState<VerificationAvailableMethods>(null)
+  const [status, setStatus] = useState<
+    "idle" | "pending" | "success" | "error"
+  >("idle")
+  const [error, setError] = useState(null)
+
+  const doFetch = useCallback(async () => {
+    try {
+      const fetch = await fetchVerificationAvailableMethods(app.id)
+      setVerificationMethods(fetch)
+      setStatus("success")
+    } catch (err) {
+      setStatus("error")
+      setError(err)
+    }
+  }, [app.id])
+
+  useEffect(() => {
+    doFetch()
+  }, [app.id, doFetch])
+
+  if (["pending", "idle"].includes(status)) {
+    return <Spinner size="m" />
+  }
+
+  let content: ReactElement
+  if (error) {
+    content = <p>{t(error)}</p>
+  } else {
+    content = (
+      <div className="flex flex-col gap-2 rounded-2xl bg-bgColorSecondary p-2">
+        {verificationMethods?.methods.map((methodType) => {
+          if (methodType.method === "website") {
+            return (
+              <Disclosure key={methodType.method}>
+                {({ open }) => <div>{methodType.website}</div>}
+              </Disclosure>
+            )
+          }
+          if (methodType.method === "login_provider") {
+            return (
+              <Disclosure key={methodType.method}>
+                {({ open }) => (
+                  <div>
+                    {methodType.login_name} {methodType.login_provider}
+                  </div>
+                )}
+              </Disclosure>
+            )
+          }
+        })}
+        <Button onClick={() => verifyApp(app.id)}>{t("verify-app")}</Button>
```

Comment:
The `verify` endpoint is only for "login_provider" verification. Website verification is automatic.

Reply:
Moved inside that code path